### PR TITLE
[openmp] On Windows, fix standalone cmake build

### LINKED ISSUE #80117: BUG: `openmp/cmake/HandleOpenMPOptions.cmake` uses CMake function `append` without defining it
While [building](https://github.com/conda-forge/openmp-feedstock/pull/113) openmp 18.1.0-rc1 on windows, I get:
```
CMake Error at cmake/HandleOpenMPOptions.cmake:47 (append):
  Unknown CMake command "append".
```

This is due to https://github.com/llvm/llvm-project/blame/release/18.x/openmp/cmake/HandleOpenMPOptions.cmake using `append` since https://github.com/llvm/llvm-project/commit/15fdc7646ca24506ccb6a59861da8f3b52ab031c, but not defining it -- in contrast to `append_if` which explicitly duplicates the definition from [`HandleLLVMOptions.cmake`](https://github.com/llvm/llvm-project/blob/release/18.x/llvm/cmake/modules/HandleLLVMOptions.cmake):
https://github.com/llvm/llvm-project/blob/400a02bb281ba0de79211ab53d931e44dd76733f/openmp/cmake/HandleOpenMPOptions.cmake#L1-L10

I don't know why this file does not (or cannot) do `include(HandleLLVMOptions)`, but if that's not an option, then it needs to similarly vendor the definition of `append`.

CC @aganea

### PR DESCRIPTION
**Pull request by aganea**
This fixes: https://github.com/llvm/llvm-project/issues/80117

### CODE REVIEW

File: openmp/cmake/HandleOpenMPOptions.cmake [hunk 1]
```diff
@@ -44,9 +44,9 @@ append_if(OPENMP_HAVE_DATA_SECTIONS "-fdata-sections" CMAKE_C_FLAGS CMAKE_CXX_FL
 
 if (MSVC)
   # Disable "warning C4201: nonstandard extension used: nameless struct/union"
-  append("-wd4201" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+  append_if(TRUE "-wd4201" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
```
Comment (aganea) [inline openmp/cmake/HandleOpenMPOptions.cmake:47]: Using `append_if(TRUE ...)` because on MSVC silencing unknown warnings is a silent action.

Comment (shiltian) [reply openmp/cmake/HandleOpenMPOptions.cmake:47]: Can we just use `list(APPEND ...)` here?

**Comment by shiltian** [comment]
`append_if(TRUE...)` looks like a "hack" to me. Can we add the def of `append` and implement `append_if` with that?

**Comment by aganea** [comment]
As suggested.

File: openmp/cmake/HandleOpenMPOptions.cmake [hunk 2]
```diff
@@ -9,6 +9,14 @@ if (NOT COMMAND append_if)
   endfunction()
```
Comment (shiltian) [inline openmp/cmake/HandleOpenMPOptions.cmake:9]: Then here we can just call `append`

**Review by jhuber6 — APPROVED** [review]

**Pull request merged**